modules/basicfuncs/list-funcs.c

```diff
@@ -70,6 +72,300 @@ tf_list_append(LogMessage *msg, gint argc, GString *argv[], GString *result)
 
 TEMPLATE_FUNCTION_SIMPLE(tf_list_append);
 
+typedef enum _ListSearchMode
+{
+  LSM_LITERAL = 0,
+  LSM_PREFIX,
+  LSM_SUBSTRING,
+  LSM_GLOB,
+  LSM_PCRE,
+} ListSearchMode;
+
+typedef struct _ListSearchPattern
+{
+  GString *keyword;
+  GPatternSpec *glob;
+  GString *keyword_reversed;
+  pcre *pcre;
+  pcre_extra *pcre_extra;
+} ListSearchPattern;
+
+static gboolean
+list_search_pattern_prepare_glob(ListSearchPattern *self)
+{
+  self->glob = g_pattern_spec_new(self->keyword->str);
+  self->keyword_reversed = scratch_buffers_alloc();
+  g_string_assign(self->keyword_reversed, self->keyword->str);
+  g_strreverse(self->keyword_reversed->str);
+
+  return TRUE;
+}
+
+static gboolean
+list_search_pattern_prepare_pcre(ListSearchPattern *self)
+{
+  const gchar *errptr;
+  gint erroffset;
+  gint rc;
+
+  self->pcre = pcre_compile2(self->keyword->str, PCRE_ANCHORED, &rc, &errptr, &erroffset, NULL);
+  if (!self->pcre)
+    {
+      msg_error("Error while compiling regular expression",
+                evt_tag_str("regular_expression", self->keyword->str),
+                evt_tag_str("error_at", &self->keyword->str[erroffset]),
+                evt_tag_int("error_offset", erroffset),
+                evt_tag_str("error_message", errptr),
+                evt_tag_int("error_code", rc));
+      return FALSE;
+    }
+  self->pcre_extra = pcre_study(self->pcre, 0, &errptr);
```

Comment:
Please use `PCRE_STUDY_JIT_COMPILE` to enable JIT whenever possible.

Reply:
Thanks, done!